roles/lib_openshift/library/oc_obj.py

```diff
@@ -1539,12 +1539,9 @@ def run_ansible(params, check_mode=False):
         # Delete
         ########
         if state == 'absent':
-            # if we were passed a name, verify its not in our results
-            if params['name'] is not None and not Utils.exists(api_rval['results'], params['name']):
-                return {'changed': False, 'state': state}
-
-            # verify results are empty for the selector
-            if params['selector'] is not None and len(api_rval['results']) == 0:
+            # verify its not in our results
+            if (params['name'] is not None or params['selector'] is not None) and \
```

Comment:
I actually prefer the logic in 2 separate steps as its more clear to me what's happening.  

I won't block on this.